### PR TITLE
Handle all exceptions within functions

### PR DESCRIPTION
**Pull request by MarioIvancik**
## Release Notes

- **New Features**
	- Updated API endpoint for improved functionality.
	- Enhanced error handling across multiple API methods.
	- Incremented version number to 2.0.0.

- **Bug Fixes**
	- Adjusted return values for API methods to provide clearer feedback on success or failure.

- **Documentation**
	- Updated test execution instructions for clarity.

- **Tests**
	- Modified test cases to reflect changes in expected API behavior.

### CODE REVIEW

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 1]
```diff
@@ -129,27 +141,46 @@ void FleetApiClient::sendStatus(const std::string &statusJson, const StatusType
...
-	return availableDevicesRequest.get();
+	std::shared_ptr<model::AvailableDevices> availableDevices {};
+	auto rc = ReturnCode::OK;
+
+	try {
+		const auto availableDevicesRequest = moduleApi_->availableDevices(companyName_, carName_, moduleIdBoost);
+		availableDevices = availableDevicesRequest.get();
+	} catch(std::exception &) {
+		rc = ReturnCode::FAILED;
+	}
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:184]: The http client should not log anything

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 2]
```diff
@@ -72,13 +73,14 @@ std::pair<std::vector<std::shared_ptr<model::Car>>, FleetApiClient::ReturnCode>
...
 {
-	const auto commandsRequest = deviceApi_->listCommands(companyName_, carName_, since.value_or(0), wait.value_or(false));
 	std::vector<std::shared_ptr<model::Message>> commands {};
 	auto rc = ReturnCode::OK;
 
 	try {
+		const auto commandsRequest = deviceApi_->listCommands(companyName_, carName_, since.value_or(0), wait.value_or(false));
 		commands = commandsRequest.get();
 	} catch(std::exception &) {
+		rc = ReturnCode::FAILED;
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:83]: The http client should not log anything

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 3]
```diff
@@ -91,13 +93,14 @@ std::pair<std::vector<std::shared_ptr<model::Message>>, FleetApiClient::ReturnCo
...
 {
-	const auto statusesRequest = deviceApi_->listStatuses(companyName_, carName_, since.value_or(0), wait.value_or(false));
 	std::vector<std::shared_ptr<model::Message>> statuses {};
 	auto rc = ReturnCode::OK;
 
 	try {
+		const auto statusesRequest = deviceApi_->listStatuses(companyName_, carName_, since.value_or(0), wait.value_or(false));
 		statuses = statusesRequest.get();
 	} catch(std::exception &) {
+		rc = ReturnCode::FAILED;
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:103]: The http client should not log anything

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 4]
```diff
@@ -129,27 +141,46 @@ void FleetApiClient::sendStatus(const std::string &statusJson, const StatusType
...
 	statuses.push_back(messagePtr_);
 
-	const auto statusesRequest = deviceApi_->sendStatuses(companyName_, carName_, statuses);
-	statusesRequest.wait();
+	try {
+		const auto statusesRequest = deviceApi_->sendStatuses(companyName_, carName_, statuses);
+		statusesRequest.wait();
+	} catch(std::exception &) {
+		return ReturnCode::FAILED;
+	}
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:163]: The http client should not log anything

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 5]
```diff
@@ -53,13 +53,14 @@ void FleetApiClient::setDeviceIdentification(const cxx::DeviceID &deviceId) cons
...
 {
-	const auto carsRequest = carApi_->availableCars(wait.value_or(false), since.value_or(0));
 	std::vector<std::shared_ptr<model::Car>> cars {};
 	auto rc = ReturnCode::OK;
 
 	try {
+		const auto carsRequest = carApi_->availableCars(wait.value_or(false), since.value_or(0));
 		cars = carsRequest.get();
 	} catch(std::exception &) {
+		rc = ReturnCode::FAILED;
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:63]: The http client should not log anything

File: src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp [hunk 6]
```diff
@@ -107,20 +110,29 @@ std::pair<std::vector<std::shared_ptr<model::Message>>, FleetApiClient::ReturnCo
...
 	commands.push_back(messagePtr_);
 
-	const auto commandsRequest = deviceApi_->sendCommands(companyName_, carName_, commands);
-	commandsRequest.wait();
+	try {
+		const auto commandsRequest = deviceApi_->sendCommands(companyName_, carName_, commands);
+		commandsRequest.wait();
+	} catch(std::exception &) {
+		return ReturnCode::FAILED;
+	}
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply src/bringauto/fleet_protocol/http_client/FleetApiClient.cpp:131]: The http client should not log anything

File: example/main.cpp [hunk 7]
```diff
@@ -8,7 +8,7 @@ std::unique_ptr<FleetApiClient> fleetApiClient;
 
 void createFleetApiClient() {
 	FleetApiClient::FleetApiClientConfig facConfig {
-		.apiUrl = "http://localhost:8080",
+		.apiUrl = "http://localhost:8080/v2/protocol",
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply example/main.cpp:11]: This is only an example of an implementation

File: example/main.cpp [hunk 8]
```diff
@@ -26,65 +26,73 @@ void createFleetApiClient() {
 }
 
 void getCars() {
-	try {
-		auto cars = fleetApiClient->getCars();
+	auto [cars, rc] = fleetApiClient->getCars();
+	if (rc != FleetApiClient::ReturnCode::OK) {
+		std::cerr << "getCars() function failed" << std::endl;
+		return;
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply example/main.cpp:32]: This is only an example of an implementation

File: example/main.cpp [hunk 9]
```diff
@@ -26,65 +26,73 @@ void createFleetApiClient() {
...
 
-		std::cout << "Cars:" << std::endl;
-		for(const auto &car: cars.first) {
-			std::cout << "  " << car->getCompanyName() << " / " << car->getCarName() << std::endl;
-		}
-	} catch(const std::exception &e) {
-		std::cerr << e.what() << std::endl;
+	std::cout << "Cars:" << std::endl;
+	for(const auto &car: cars) {
+		std::cout << "  " << car->getCompanyName() << " / " << car->getCarName() << std::endl;
```
[the remark this answers is not in the document]
Comment (MarioIvancik) [reply example/main.cpp:37]: This is only an example of an implementation

**Review by koudis — APPROVED** [review]

**Pull request merged**